variables.tf

```diff
@@ -57,9 +62,9 @@ variable "agent_name" {
   nullable    = false
 }
 
-variable "agent_resource_group_id" {
+variable "agent_resource_group_name" {
```

Comment:
We should revert this back to ID for consistency with other modules. If provider requires name, lets use our resource group module to convert and pass to provider that way. Resource group names can be updated. IDs are static.

Reply:
Generally an end user provides the resource group name and then in examples we pass `id / name` as per the requirement to the root module. For schematics agent - provider requires name only.

If user provides -> RG name, convert to RG ID -> Pass to root module, 
In root module -> convert RG ID to RG name and then pass RG name -> provider
Why to do all this? instead let user provide resource group name only and then internally we can use either name/id.

Shall I rename `agent_resource_group_name` -> `agent_resource_group` to avoid any confusion?

Reply:
Neither provider or our module supports that if RG ID is passed, it returns RG name.
Hence can not do. Creating a new issue to address but proceeding with RG name for this module at this moment.